pyproject.toml

```diff
@@ -44,12 +44,13 @@ ctc = [
     "tifffile>=2024.10",
 ]
 rx = ["rustworkx>=0.16.0"]
+df = ["pandas>=2","pandas>=2.3; python_version >= '3.12'"]
```

Comment:
@tlambert03 I got into a bit of a fight with CI trying to get the lowest-direct versions of the tests passing with pandas. Do these pins seem reasonable to you?

Reply:
ah... you were probably realizing that you need to be a bit more involved (than you'd like to have to be) in picking the lowest supported pandas based on which wheels are on pypi right?  There's a uv issue for that somewhere... will try to find.  But I often just end up doing what you've done here

Reply:
Yup that is exactly what happened. I'm fine with leaving it as is knowing that you have gone down a similar rabbit hole before 😂 

Good call about df vs. pandas. I don't have any intention of supporting a different pandas-like backend so I'll switch it to pandas